chemprop/nn/transforms.py

```diff
@@ -0,0 +1,63 @@
+from abc import abstractmethod
+
+import torch
+from numpy.typing import ArrayLike
+from sklearn.preprocessing import StandardScaler
+from torch import Tensor, nn
+
+from chemprop.data.collate import BatchMolGraph
+
+
+class TensorTransformBase(nn.Module):
+    def __init__(self, mean: ArrayLike, scale: ArrayLike):
+        super().__init__()
+
+        self.register_buffer("mean", torch.tensor(mean, dtype=torch.float))
+        self.register_buffer("scale", torch.tensor(scale, dtype=torch.float))
+
+    @classmethod
+    def from_standard_scaler(cls, scaler: StandardScaler):
+        return cls(scaler.mean_, scaler.scale_)
+
+    @abstractmethod
+    def forward(self, X: Tensor) -> Tensor:
+        pass
+
+
+class TensorTransform(TensorTransformBase):
+    def forward(self, X: Tensor) -> Tensor:
+        if self.training:
+            return X
+
+        return (X - self.mean) / self.scale
+
+
+class TensorUntransform(TensorTransformBase):
+    def forward(self, X: Tensor) -> Tensor:
+        if self.training:
+            return X
+
+        return X * self.scale + self.mean
+
+
+class GraphTransform(nn.Module):
+    def __init__(self, mean: ArrayLike, scale: ArrayLike, key: str, inds: ArrayLike):
+        super().__init__()
+
+        assert key in {"V", "E"}
+
+        self.transform = TensorTransform(mean, scale)
+        self.key = key
+        self.inds = inds
+
+    def from_standard_scaler(scaler: StandardScaler, key: str, inds: ArrayLike):
+        return GraphTransform(scaler.mean_, scaler.scale_, key, inds)
+
+    def forward(self, bmg: BatchMolGraph) -> BatchMolGraph:
+        if self.training:
+            return bmg
+
+        X = getattr(bmg, self.key)
+        X[:, self.inds] = self.transform(X[:, -self.inds])
+
+        return bmg
```

Comment:
I was envisioning that a `GraphTransform` is _composed_ of two `TensorTransform`s, one for `V` and one for `E`, so we don't need to worry about variably getting the correct attribute (like you do with `self.key`). I see that you need some notion of the indices to scale (via `self.inds`), but perhaps this can also be subsumed into the `TensorTransform`?

Reply:
or you can avoid using some notion of indexing and instead _mock_ the scaling by setting the mean and scale for these _unscaled_ indices to 0 and 1, respectively?